fml/platform/fuchsia/message_loop_fuchsia.cc

```diff
@@ -6,33 +6,57 @@
 
 #include <lib/async-loop/default.h>
 #include <lib/async/cpp/task.h>
-#include <lib/zx/time.h>
 
 namespace fml {
 
 MessageLoopFuchsia::MessageLoopFuchsia()
-    : loop_(&kAsyncLoopConfigAttachToCurrentThread) {}
+    : loop_(&kAsyncLoopConfigAttachToCurrentThread), running_(false) {
+  auto status =
+      zx::timer::create(ZX_TIMER_SLACK_LATE, ZX_CLOCK_MONOTONIC, &timer_);
+  FML_CHECK(status == ZX_OK);
+  timer_wait_ = std::make_unique<async::Wait>(timer_.get(), ZX_TIMER_SIGNALED);
+
+  auto wait_handler = [this](async_dispatcher_t* dispatcher, async::Wait* wait,
+                             zx_status_t status,
+                             const zx_packet_signal_t* signal) {
+    timer_.cancel();
+    RunExpiredTasksNow();
+    if (status != ZX_ERR_CANCELED) {
+      FML_CHECK(signal->observed & ZX_TIMER_SIGNALED);
+      auto result = wait->Begin(loop_.dispatcher());
+      FML_CHECK(result == ZX_OK);
+    }
+  };
+
+  timer_wait_->set_handler(wait_handler);
+  status = timer_wait_->Begin(loop_.dispatcher());
+  FML_CHECK(status == ZX_OK);
+}
 
 MessageLoopFuchsia::~MessageLoopFuchsia() = default;
 
 void MessageLoopFuchsia::Run() {
-  loop_.Run();
+  running_ = true;
+
+  while (running_) {
+    auto status = loop_.Run(zx::time::infinite(), true);
```

Comment:
Minor nit: It's hard to tell what `true` means here. When there is such ambiguity, we generally like to comment what the argument means. See https://github.com/flutter/engine/blob/400ed7cebc6443cccc869eddea08c87c5af16e27/runtime/dart_isolate.cc#L598.